eeps/eep-0064.md

```diff
@@ -0,0 +1,354 @@
+        This text
+        has no indentation
+        """,
+    
+    "    This text\n    has indentation" = """
+            This text
+            has indentation
+        """,
+    
+    "  This text\nhas an indented first line" =
+        """
+          This text
+        has an indented first line
+        """,
+    
+    """
+    This is a syntax error (incorrect indentation)
+        """,
+
+    """ This is a syntax error
+    (non-white-space on start line)
+    """,
+
+    """
+    This will probably be a syntax error
+    since no ending line can be found"""
+
+### Backwards incompatibility
+
+This is valid today:
+
+    X = """
+        X
+        """
+
+It is equivalent to:
+
+    X = "" "
+        X
+        " ""
+
+Which is equivalent to:
+
+    X = "
+        X
+        "
+
+Which is equivalent to:
+
+    X = "\n    X\n"
+
+But with the suggested MIV Strings the first
+code snippet would instead be equivalent to:
+
+    X = "X"
+
+Also, this is valid today:
+
+    X = """ xxx
+      X
+        """
+
+But according to this EEP it would be two syntax errors:
+
+1. The start line has got non-white-space after `"""`.
+2. The first content line has incorrect indentation.
+
+There are many other similar constructions that also
+would be syntax errors.
+
+* It is far from likely that anyone has deliberately
+  used `"""` in source code to mean an empty string
+  concatenated to another string.
+* Most today allowed combinations with `"""` will cause
+  syntax errors.  Only a few will have a subtly changed
+  behaviour (string content).
+* Users can simply grep for `"""` in their source code.
+  Causing the same sequence e.g through macros would
+  be harder to find, but the worst problem would not
+  be new syntax errors (hard to miss), but changed
+  behaviour.  And the changed behaviour would be
+  a slightly different string content.
+
+Therefore, it should be very unlikely that anyone
+encounters a real backwards incompatibility problem
+from the suggestions in this EEP.
+
+### Quoting of `"""`
+
+In the rules above there is no possibility to have `"""`
+first on a line in a MIV String.
+
+This would be allowed:
+
+    -doc """
+        A MIV String starts with: """
+        and ends with: """
+        """.
+
+As long as `"""` isn't first on a line.
+
+It would be possible to work around in Erlang code:
+
+    X = """
+        A MIV String starts with:
+        ""
+        """ """
+        "
+        and ends with:
+        ""
+        """ "\""
+
+That is ugly, and it is not possible in a documentation
+attribute where string concatenation isn't allowed.
+
+We can either ignore the problem since it is only
+when placed first on a line that `"""` is a problem,
+or we can use the GitHub [Markdown][] trick to allow
+3 or more start characters and matching end characters
+so this would be valid:
+
+    X = """"
+        A MIV String starts with:
+        """
+        and ends with:
+        """
+        """"
```

Comment:
I liked the four double quotes idea.
I've updated [my PR](https://github.com/erlang/otp/pull/7313) with this change. Now this is possible:
```bash
1> """"
.. ```erlang
.. foo() ->
..     """
..     foo
..     bar
..     """.
.. ```
.. """".
"```erlang\nfoo() ->\n    \"\"\"\n    foo\n    bar\n    \"\"\".\n```\n"
```

Reply:
C# has this three-or-more double quotes syntax too.

Reply:
@williamthome: Note that the idea is to allow 3 or more, not 3 or 4...

Reply:
@RaimoNiskanen Oh, nice! I'll change the code to support 3+ double quotes.

Reply:
The PR is updated and now 3+ double quotes are supported.

Reply:
I am a bit late to the party but I like the idea a lot too. This and automatically stripping the last line are elegant solutions to the problems presented. :+1: